# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: netwerk/test/unit/test_httpssvc_https_upgrade.js
# Commit: 4309b577f25a
# Full Hash: 4309b577f25aafdee2af81caf35faae93d05e75b
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2021-01-29 16:01:10
# Description:
#   Bug 1688945 - Make sure we don't query HTTPS RR when we already have one r=necko-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D103348
# ==============================================================================

diff -r 5efe821951f9 -r 4309b577f25a netwerk/test/unit/test_httpssvc_https_upgrade.js
--- a/netwerk/test/unit/test_httpssvc_https_upgrade.js	Thu Jan 28 18:10:56 2021 +0000
+++ b/netwerk/test/unit/test_httpssvc_https_upgrade.js	Thu Jan 28 19:34:41 2021 +0000
@@ -62,6 +62,11 @@
     "https://foo.example.com:" + h2Port + "/httpssvc_as_altsvc"
   );
 
+  Services.prefs.setBoolPref(
+    "network.dns.use_https_rr_for_speculative_connection",
+    true
+  );
+
   // The moz-http2 cert is for foo.example.com and is signed by http2-ca.pem
   // so add that cert to the trust list as a signing cert.  // the foo.example.com domain name.
   const certdb = Cc["@mozilla.org/security/x509certdb;1"].getService(
@@ -88,6 +93,7 @@
   prefs.clearUserPref("network.trr.clear-cache-on-pref-change");
   prefs.clearUserPref("network.dns.upgrade_with_https_rr");
   prefs.clearUserPref("network.dns.use_https_rr_as_altsvc");
+  prefs.clearUserPref("network.dns.use_https_rr_for_speculative_connection");
 });
 
 class DNSListener {
